esrally/utils/net.py

```diff
@@ -146,8 +194,9 @@ def download(url, local_path, expected_size_in_bytes=None, progress_indicator=No
     """
     tmp_data_set_path = local_path + ".tmp"
     try:
-        if url.startswith("s3"):
-            expected_size_in_bytes = download_s3(url, tmp_data_set_path, expected_size_in_bytes, progress_indicator)
+        scheme = urllib3.util.parse_url(url).scheme
+        if scheme in ["s3", "gs"]:
```

Comment:
I wonder whether we should start encapsulating this now that we add a third scheme. We could implement this as a `Downloader` that supports certain schemes, register them upon startup and implement `def supports(self, url: str) -> bool` to check to which `Downloader` to pass a URL instead of spreading this logic in multiple places. Wdyt?

Reply:
As discussed offline, we'll defer this for now, but certainly look at it again, if we are to extend to another scheme.